### ISSUE TITLE
[ENH] Adapt `plot_series` to accept numpy arrays and list of numpy

### ISSUE THREAD

**Issue by hadifawaz1999** [aeon-toolkit/aeon#1036]
### Describe the feature or idea you want to propose

For now the function `plot_series` accepts only pd.Series, which is not the best approach because if someone works with numpy than the code will crash. 

### Describe your proposed solution

It should be adapted to work with 3d numpy, 2d numpy and list of 2d numpy

**Comment by MatthewMiddlehurst** [comment]
I think `plot_series` only work with a singular time series, though that may be wrong. May need a separate functions for multi-dimensional series/collections.

**Comment by hadifawaz1999** [comment]
from what i understand in the docs @MatthewMiddlehurst it can work on a collection

**Comment by Cyril-Meyer** [comment]
> from what i understand in the docs @MatthewMiddlehurst it can work on a collection

It does :

```python
y1 = load_airline()
y2 = y1 ** 0.99
y3 = y1 ** 0.97

fig, ax = plot_series(y1, y2, y3)
plt.show()
```

![image](https://github.com/user-attachments/assets/4266e0e0-8d3b-47dd-8ebd-8a309399e07e)

> It should be adapted to work with 3d numpy, 2d numpy and list of 2d numpy

Working with 1d numpy array should be fine, we can add a check like `if numpy and 1d: x=pd.Series(x)` :

```python
fig, ax = plot_series(pd.Series(np.array(y1)), pd.Series(np.array(y2)), pd.Series(np.array(y3)))
plt.show()
```

![image](https://github.com/user-attachments/assets/45ff4fd7-1a0d-4270-b020-2da9fabf3d7c)

Working with 2d numpy array should also be fine, we can add a check add a conversion function : 
```python
def pd_s_to_np_2d(series):
    index_values = series.index.values
    series_values = series.values
    array_2d = np.column_stack((index_values, series_values))
    return array_2d

def np_2d_to_pd_s(array_2d):
    index_values = array_2d[:, 0]
    series_values = array_2d[:, 1]
    series = pd.Series(series_values, index=index_values)
    return series

fig, ax = plot_series(np_2d_to_pd_s(pd_s_to_np_2d(y1)), np_2d_to_pd_s(pd_s_to_np_2d(y2)), np_2d_to_pd_s(pd_s_to_np_2d(y3)))
plt.show()

```

![image](https://github.com/user-attachments/assets/208086a3-466f-4a75-95f2-0a1b4d9de2df)

@hadifawaz1999 
For 3D and 2D list, we can do something also, but I don't think it's a good idea to have at the same time the current system of `*series` being a variable length argument and accepting 2D list or 3D array.
I don't know which is best for now, and I'm interested in your opinion. I've thought about the following solution (ranked by better to worse in my opinion)
1. Add a new function `plot_series_np` for NumPy (including 3D and 2D list and others) but also improve plot_series with simple numpy to series conversion and call to this new function. In real life, `plot_series` will call `plot_series_np` if user don't use it.
2. Allow multiple 2D list or 3D array and just concatenate them or just show the first.
3. Add a new function `plot_series_np` for NumPy plots but don't change anything in `plot_series`.
4. I just forget it while writing...

**Comment by hadifawaz1999** [comment]
Thanks @Cyril-Meyer , am not sure if we were planning on keeping pd.Series stuff in visualization, @TonyBagnall can you confirm ?

If not maybe we can allow either list of 2d numpy for unequal length series, or 3d numpy for equal length, or a single 2d numpy for one series, kinda how we handle pairwise plots in distances module. @dguijo any ideas on this ?

**Comment by MatthewMiddlehurst** [comment]
I would be better to have a separate function for collections, there is one already. The plot functions uses the index from `pd.Series`, which is why thats the input type.

**Comment by TonyBagnall** [comment]
fixed by #2279, made it only a series plotter
